src/core/object/controller/controller.ts

```diff
@@ -127,6 +130,13 @@ export default class Controller {
 				type: 'setEditState',
 				fn: (isEditing) => {
 					this.isEditing = isEditing;
+					if (isEditing) {
+						clearTimeout(this.setNotEditingTimeout);
+						this.setNotEditingTimeout = setTimeout(() => {
+							this.isEditing = false;
+							this.eventEmitter.emit('updateEditStatus', false);
+						}, 5000);
+					}
```

Comment:
Shoudl we return here? or is it okays that we call `this.eventEmitter.emit('updateEditStatus', isEditing);`?

Reply:
It is not only ok, it should be called. I'll bump the line up to make it a bit clearer

Reply:
Basically the emitter outside the timeout will instantly set controller to not be in edit mode if the user cancels the edit within the popup or finishes the edit, while the timeout takes care of the case where the user just clicks outside the popup while editing